applications/machine_learning/README.rst

```diff
@@ -468,16 +429,43 @@ Application internal modules
-`Module event tables`_ for some modules include extensive lists of source and sink modules.
-These are valid for events that have many listeners or sources, and are gathered on the :ref:`nrf_machine_learning_app_event_rel_modules` subpage.
-
-.. toctree::
-   :maxdepth: 1
-   :caption: Subpages:
 
-   doc/event_rel_modules.rst
-   doc/main.rst
-   doc/led_state.rst
+The nRF Machine Learning application uses the following modules available in CAF:
+
+* :ref:`caf_ble_adv`
+* :ref:`caf_ble_state`
+* :ref:`caf_buttons`
+* :ref:`caf_click_detector`
+* :ref:`caf_leds`
+* :ref:`caf_sensor_sampler`
+
+More information about every used CAF module and its configuration details are available on the linked subpage of the module.
+
+The nRF Machine learning application uses also the following dedicated application modules:
+
+* ``ei_data_forwarder_bt_nus``
+  The module forwards the sensor readouts over NUS to a Bluetooth connected central.
+  The sensor data is forwarded only if the connection is secured and connection interval is within the limit defined by :option:`CONFIG_BT_PERIPHERAL_PREF_MAX_INT` and :option:`CONFIG_BT_PERIPHERAL_PREF_MAX_INT`.
+* ``ei_data_forwarder_uart``
+  The module forwards the sensor readouts over UART.
+* ``led_state``
+  The module displays the application state using LEDs.
+  The LED effects used to display state of the data forwarding, machine learning results and simulated signal are defined in :file:`led_state_def.h` file located in the application configuration directory.
+  Common LED effects are used to represent the machine learning results and the simulated sensor signal.
+* ``ml_runner``
+  The module uses :ref:`ei_wrapper` API to control running the machine learning model.
+  The module provides the prediction results using :c:struct:`ml_result_event`.
+  The module runs the machine learning model and provides results only if there is an active subsriber.
+  An application modules can inform that it is actively listening for results using an :c:struct:`ml_result_signin_event`.
+* ``ml_state``
+  The module controls switching between running the machine learning model and forwarding the data.
+  The change is triggered by long press of the button defined in the module's configuration.
+* ``sensor_sim_ctrl``
+  The module controls parameters of the generated simulated sensor signal.
+  The module switches between predefined sets of parameters for the simulated signal.
+  Parameters of the generated signals are defined by the :file:`sensor_sim_ctrl_def.h` file located in the application configuration directory.
+
+.. warning::
+   The ``ei_data_forwarder_bt_nus`` and ``ei_data_forwarder_uart`` stop forwarding the sensor readouts if they receive a :c:struct:`sensor_event` that cannot be forwarded and needs to be dropped.
+   This could happen, for example, if the selected sensor sampling frequency is too high for used implementation of the Edge Impulse data forwarder.
```

Comment:
```suggestion
   This could happen, for example, if the selected sensor sampling frequency is too high for the used implementation of the Edge Impulse data forwarder.
```